sphinx_gallery/utils.py

```diff
@@ -36,3 +36,15 @@ def __init__(self):  # noqa: D102
 
     def __del__(self):  # noqa: D105
         rmtree(self._path, ignore_errors=True)
+
+
+def replace_py_ipynb(fname):
+    """Replace .py extension in filename by .ipynb"""
+    fname_prefix, extension = fname.rsplit('.', 1)
```

Comment:
how about `os.path.splitext` instead: https://docs.python.org/2/library/os.path.html#os.path.splitext

Reply:
(it seems like the functionality would be the same though...so I don't think it's blocking this PR)

Reply:
Done.